src/services/TestTypesService.ts

```diff
@@ -51,20 +51,25 @@ export class TestTypesService {
         testCodes = testCodes.filter((testCode: ITestType) => { // filter by vehicleType if present in DB, otherwise skip
           return testCode.forVehicleType ? testCode.forVehicleType === filterExpression.vehicleType : true;
         });
-        testCodes = testCodes.filter((testCode: ITestType) => { // filter by vehicleSize if present in DB, otherwise skip
+        testCodes = testCodes.filter((testCode: ITestType) => { // filter by vehicleSize if present in DB & in request, otherwise skip
           return (testCode.forVehicleSize && filterExpression.vehicleSize) ? testCode.forVehicleSize === filterExpression.vehicleSize : true;
         });
-        testCodes = testCodes.filter((testCode: ITestType) => { // filter by vehicleConfiguration if present in DB, otherwise skip
-          return testCode.forVehicleConfiguration ? testCode.forVehicleConfiguration === filterExpression.vehicleConfiguration : true;
+        testCodes = testCodes.filter((testCode: ITestType) => { // filter by vehicleConfiguration if present in DB & in request, otherwise skip
```

Comment:
these should probably just be chained, rather than thrown out to a variable every time, but not a big deal